### ISSUE TITLE
After bundle exec jekyll serve uploaded cover image is not copied to _site folder. 

### ISSUE THREAD

**Issue by venkimithu** [strapi/jekyll-strapi#10]
```
---
layout: default
---
<img src="{{ page.document.cover[0].url }}">
<h1>{{page.document.title}}</h1>
<span class="post-meta">{{ page.document.createdAt | date_to_string }} by <a
        href="/authors/{{ page.document.author.username }}">{{ page.document.author.username }}</a></span>
{{ page.document.content | markdownify }}
```

Not able to copy uploaded cover image to the _site folder. Please help with the same.

**Comment by tobymurray** [comment]
I also ran into this. Following https://strapi.io/blog/building-a-static-website-using-jekyll-and-strapi I don't see any media being accessible in Jekyll at all.

**Comment by jbrew30** [comment]
They removed the `cover` from this whole [tutorial](https://strapi.io/blog/building-a-static-website-using-jekyll-and-strapi). I had high hope for this, and it was quite enjoyable to learn up until the point you figure out the media library is off-limits.

**Comment by Mcastres** [comment]
Hey ! Thanks for reporting the issue. I'm currently updating old tutorials and this one is on the list! Sorry for the inconvenience, it will be updated soon :)

**Comment by jbrew30** [comment]
@Mcastres I do appreciate the update and the speed at which it was updated. Adding the template data does help in understanding the data modeling a bit better within Strapi. But the rewrite doesn't address the original issue of not being able to add image assets. When you run `bundle exec Jekyll serve` the contents of the backend/public/uploads are not packaged and added within the `_site` folder. 

Also within the `_config.yaml` the articles type needs an "s" at the end of `type: articles`. If it's not applied you will get an error. 

<img width="576" alt="Screen Shot 2021-02-19 at 1 57 23 PM" src="https://user-images.githubusercontent.com/22452733/108548924-7bb6e780-72ba-11eb-8b86-eee5389cfe6c.png">


Needs to look like this

<img width="594" alt="Screen Shot 2021-02-19 at 1 58 32 PM" src="https://user-images.githubusercontent.com/22452733/108548995-95582f00-72ba-11eb-99a5-860c243af3f2.png">

**Comment by Mcastres** [comment]
Sorry, it was a typo actually, you can find the repo for this tutorial here: https://github.com/strapi/jekyll-blog
I will investigate on the image assets not being added !

However, I'm not an expert on this, but you should be able to retrieve your assets using the Strapi API_URL.

```
gem install jekyll-environment-variables
```

```
plugins:
  - jekyll-environment-variables
```

-  Create an `API_URL` containing the Strapi URL: https://localhost:1337

```
---
layout: default
---
<img src="{{ site.env.API_URL | append: page.document.cover[0].url }}">
<h1>{{page.document.title}}</h1>
<span class="post-meta">{{ page.document.createdAt | date_to_string }} by <a
        href="/authors/{{ page.document.author.username }}">{{ page.document.author.username }}</a></span>
{{ page.document.content | markdownify }}
```

**Comment by jbrew30** [comment]
I'm not an expert either. I just started using Jekyll a week ago. Could you elaborate a bit more on "Create an `API_URL` containing the Strapi URL: https://localhost:1337. Do I need to create a .env file and add `API_URL: "http://localhost:1337/"`? Or do I add `API_URL: "http://localhost:1337/"` to my `_config.yml` file?

 Also, shouldn't it be  `<img src="{{ site.env.API_URL + page.document.image[0].url }}">`  since that's what its named here?



<img width="555" alt="Screen Shot 2021-02-22 at 4 41 44 PM" src="https://user-images.githubusercontent.com/22452733/108774176-51b92b80-752d-11eb-8ed8-b08b99d37cb7.png">

I really hate to sound nitpicky; I really want this to work and maybe help others down the ride if they find this.

**Comment by Mcastres** [comment]
Yes, if you named it `image` then it will be `<img src="{{ site.env.API_URL | append:  page.document.image[0].url }}">` but I used cover since @venkimithu used it.

You can:

  - `export API_URL=http://localhost:1337`
  - Create a `_plugins/env.rb` containing the following code:
  
```
module Jekyll
  class EnvVariables < Generator
    def generate(site)
      site.config['env'] = {}
      ENV.each_pair { |k, v| site.config['env'][k] = v }
    end
  end
end
```

This will fetch all your env variable and make them available in `site.env`

  - Then:  

```
<img src="{{ site.env.API_URL | append: page.document.image[0].url }}">
```

**Comment by jbrew30** [comment]
Thanks, @Mcastres. I started from scratch and used this repo ( https://github.com/strapi/jekyll-blog ) and was able to bring in the path to the image. 

<img width="1387" alt="Screen Shot 2021-02-25 at 9 54 01 AM" src="https://user-images.githubusercontent.com/22452733/109172776-244cc780-7751-11eb-996e-a30bcbd76cab.png">

The contents of the upload folder is not added to the `_sites` folder, but that is remedied by just dragging the uploads folder into the `frontend` folder. 

<img width="842" alt="Screen Shot 2021-02-25 at 10 07 46 AM" src="https://user-images.githubusercontent.com/22452733/109172994-59f1b080-7751-11eb-822a-dd1b3a7a3fd1.png">

This is something I can work with. Also, FYI  when you use `yarn create strapi-app backend --template https://github.com/strapi/strapi-template-blog` `image` is the name it's given when you use that template. 

If I were you guys, I would update or tell people to use the repo I used for the frontend for jeklly integration. 

Thanks again!
